test/delete.test.js

```diff
@@ -299,23 +299,3 @@ fastify.listen({ port: 0 }, err => {
     })
   })
 })
-
-// https://github.com/fastify/fastify/issues/936
-test('shorthand - delete with application/json Content-Type header and without body', t => {
```

Comment:
I understand the RFC etc.. but this stops the fastify adoption to big (legacy) companies and tools.

I would create a new issue/feature request as follow-up, to let the user to customise the new `bodylessMethods` prop

Reply:
Problem is DELETE is also in the class of methods that is permitted to have a body, so in that case DELETE would be broken for requests with with a body

The issue in that test is sending (an invalid) Content-Type header like `application/json` with an empty body, this is just an incorrect request — they should've sent something like `'{}'` or not set the Content-Type header...

But I can add a `'DELETE'` string literal here if you think this would really cause companies a headache:

https://github.com/fastify/fastify/blob/97ede717084e8c7af755e9f4772c75b64f09adb0/lib/handleRequest.js#L45-L49

```js
 if (contentLength === undefined && transferEncoding === undefined && (method === 'OPTIONS' || method === 'DELETE')) { 
   // OPTIONS (and DELETE) can have a Content-Type header without a body 
   handler(request, reply) 
   return 
 } 
```

Reply:
> The issue in that test is sending (an invalid) Content-Type header like application/json with an empty body, this is just an incorrect request 

But `null` is a valid JSON: is the body empty/missing from the request?
I thought inject submits `null` as body payload.

In this case let's skip it